includes/shortcodes/class.llms.shortcodes.blocks.php

```diff
@@ -0,0 +1,248 @@
+		 */
+		return apply_filters( 'llms_shortcode_blocks', $config );
+	}
+
+	/**
+	 * Registers shortcode blocks.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function register_blocks(): void {
+		$blocks = $this->get_config();
+
+		foreach ( $blocks as $name => $args ) {
+			$block_dir = $args['path'] ?? LLMS_PLUGIN_DIR . "blocks/$name";
+
+			if ( file_exists( "$block_dir/block.json" ) ) {
+				register_block_type(
+					$block_dir,
+					array(
+						'render_callback' => array( $this, 'render_block' ),
+					)
+				);
+			}
+		}
+	}
+
+	/**
+	 * Loads front end CSS in the editor.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function add_editor_styles(): void {
+		$plugins_dir = basename( WP_PLUGIN_DIR );
+		$plugin_dir  = basename( LLMS_PLUGIN_DIR );
+		$path        = "../../$plugins_dir/$plugin_dir/assets/css/lifterlms.min.css";
+
+		add_editor_style( $path );
+	}
+
+	/**
+	 * Enqueues editor styles.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function enqueue_editor_styles(): void {
+		if ( ! llms_is_block_editor() ) {
+			return;
+		}
+
+		$path = '/assets/css/editor.min.css';
+
+		if ( ! file_exists( LLMS()->plugin_path() . $path ) ) {
+			return;
+		}
+
+		wp_enqueue_style(
+			'llms-editor',
+			LLMS()->plugin_url() . $path,
+			array(),
+			filemtime( LLMS()->plugin_path() . $path )
+		);
+	}
+
+	/**
+	 * Shows the registration and login form in editor preview.
+	 *
+	 * @since [version]
+	 *
+	 * @param bool $hide Whether to hide the registration form.
+	 * @return bool
+	 */
+	public function show_form_preview( bool $hide ): bool {
+		if ( ! defined( 'REST_REQUEST' ) || ! is_user_logged_in() ) {
+			return $hide;
+		}
+
+		global $wp;
+
+		if ( ! $wp instanceof WP || empty( $wp->query_vars['rest_route'] ) ) {
+			return $hide;
+		}
+
+		$route = $wp->query_vars['rest_route'];
+
+		if ( false !== strpos( $route, '/block-renderer/' ) ) {
+			$hide = false;
+		}
+
+		return $hide;
+	}
+
+	/**
+	 * Renders a shortcode block.
+	 *
+	 * @since [version]
+	 *
+	 * @param array    $attributes The block attributes.
+	 * @param string   $content    The block default content.
+	 * @param WP_Block $block      The block instance.
+	 * @return string
+	 */
+	public function render_block( array $attributes, string $content, WP_Block $block ): string {
+		if ( ! property_exists( $block, 'name' ) ) {
+			return '';
+		}
+
+		$name   = str_replace( 'llms/', '', $block->name );
+		$config = $this->get_config();
+		$class  = $config[ $name ]['render'][0] ?? '';
+		$method = $config[ $name ]['render'][1] ?? '';
+
+		if ( ! method_exists( $class, $method ) ) {
+			return '';
+		}
+
+		if ( method_exists( $class, 'instance' ) ) {
+			$render = array( $class::instance(), $method );
+		} else {
+			$render = array( $class, $method );
+		}
+
+		$content = $attributes['text'] ?? '';
+
+		unset( $attributes['text'] );
+
+		$shortcode = call_user_func_array( $render, array( $attributes, $content ) );
```

Comment:
what about this?

```suggestion
		$args      = array( $attributes, $content );
		$shortcode = call_user_func( $render, ...$args );
```
`call_user_func` I remember it to be faster than `call_user_func_array`.

Also I'd add a test before on if  `$config[ $name ]['render']` is a callable in that case that's already the `$render`, otherwise check the instance method existence etc.
What do you think?